lib/bootloader_setup.pm

```diff
@@ -976,35 +976,45 @@ sub tianocore_http_boot {
 }
 
 sub zkvm_add_disk {
-    my ($svirt) = @_;
-    if (my $hdd = get_var('HDD_1')) {
-        my $basename = basename($hdd);
-        my $basedir  = svirt_host_basedir();
-        my $hdd_dir  = "$basedir/openqa/share/factory/hdd";
-        my $hdd_path = $svirt->get_cmd_output("find $hdd_dir -name $basename | head -n1 | tr -d '\n'");
-        die "Unable to find image $basename in $hdd_dir" unless $hdd_path;
-        diag("HDD path found: $hdd_path");
-        if (get_var('PATCHED_SYSTEM')) {
+    my ($svirt)  = @_;
+    my $numdisks = get_var('NUMDISKS') // 1;
+    my $hdd_dir  = sprintf("%s/openqa/share/factory/hdd", svirt_host_basedir());
+    my $dev_id   = 'a';
+    for my $di (1 .. $numdisks) {
+        if (get_var('PATCHED_SYSTEM') && $dev_id eq 'a') {
             diag('in patched systems just load the patched image');
             my $name        = $svirt->name;
             my $patched_img = "$zkvm_img_path/$name" . "a.img";
             $svirt->add_disk({file => $patched_img, dev_id => 'a'});
+        } else {
+            # Copy existing disk image to local storage
+            if (get_var("HDD_$di")) {
+                my $basename = basename(get_var("HDD_$di"));
+                my $hdd_path = $svirt->get_cmd_output("find $hdd_dir -name $basename | head -n1 | tr -d '\n'");
+                $hdd_path or die "Unable to find image $basename in $hdd_dir";
+                diag("HDD path found: $hdd_path");
+
+                enter_cmd("# copying image ($basename)...");
+                if (my $size = get_var("HDDSIZEGB_$di")) {
+                    $size .= "G";
+                    $svirt->add_disk({file => $hdd_path, backingfile => 1, dev_id => $dev_id, size => $size});
+                } else {
+                    $svirt->add_disk({file => $hdd_path, backingfile => 1, dev_id => $dev_id});
+                }
+            } else {
+                # Create a new image, most likely it can be image for installation
+                # or additional optional drive for further testing
+                my $size;
+                if (get_var('ISO') && $dev_id eq 'a') {
```

Comment:
This looks a bit weird. Only if `ISO` is set, it uses `HDDSIZEGB` for the first HDD?

Reply:
Yup, [installations](https://openqa.suse.de/tests/5989991/logfile?filename=autoinst-log.txt) take the value of `HDDSIZEGB` and creates and image file.

```
[2021-05-11T00:30:45.361 CEST] [debug] tests/installation/bootloader_zkvm.pm:74 called bootloader_setup::zkvm_add_disk -> lib/bootloader_setup.pm:1002 called backend::console_proxy::__ANON__
[2021-05-11T00:30:45.362 CEST] [debug] <<< backend::console_proxy::__ANON__(wrapped_call={
  "wantarray" => undef,
  "console" => "svirt",
  "function" => "add_disk",
  "args" => [
              {
                "create" => 1,
                "dev_id" => "a",
                "size" => "40G"
              }
            ]
})
[2021-05-11T00:30:45.363 CEST] [debug] <<< backend::baseclass::run_ssh_cmd(cmd="qemu-img create /var/lib/libvirt/images/openQA-SUT-2a.img 40G -f qcow2", wantarray=1, keep_open=1, password="SECRET", username=undef, hostname="s390zp18.suse.de")
[2021-05-11T00:30:45.363 CEST] [debug] <<< backend::baseclass::run_ssh(cmd="qemu-img create /var/lib/libvirt/images/openQA-SUT-2a.img 40G -f qcow2", keep_open=1, wantarray=1, password="SECRET", hostname="s390zp18.suse.de", username=undef)
```

If `HDDSIZEGB_X` is not present for additional test drives, I am keeping the default size 4GB

Reply:
That is already covered by the common logic though, so just doing

```
$size = sprintf("%dG", get_var("HDDSIZEGB_$di", get_var('HDDSIZEGB', 4)));
```

for all new HDDs would work FWICT.

Reply:
Yup, I would prefer the code as you suggest, but the problem comes that the 4 GB default drive won't work for QEMU/pvm at least as `HDDSIZEGB` might be set by the backend implicitly.

```
../os-autoinst-distri-opensuse/os-autoinst/backend/pvm.pm:    $vars->{HDDSIZEGB} ||= 15;
../os-autoinst-distri-opensuse/os-autoinst/backend/qemu.pm:    $vars->{HDDSIZEGB} ||= 10;
../os-autoinst-distri-opensuse/os-autoinst/doc/backend_vars.asciidoc:HDDSIZEGB;integer;10;Creates HDD with specified size in GiB
```

On the other hand, 10/15 vs 4 is not that big of a deal for me.
@jlausuch ?

Reply:
Maybe that shouldn't be set in the svirt case then? Especially for sparse files this shouldn't really make much of a difference though, and when it's explicitly set the behaviour is as expected anyway.

Reply:
Yup, agreed. I will add it together with fix for https://openqa.suse.de/tests/6325018#step/image_checks/5 in os-autotinst. Thanks!